src/Microsoft.Health.Fhir.Azure/AzureAccessTokenProvider.cs

```diff
@@ -0,0 +1,48 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Services.AppAuthentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export.AccessTokenProvider;
+
+namespace Microsoft.Health.Fhir.Azure
+{
+    public class AzureAccessTokenProvider : IAccessTokenProvider
+    {
+        private readonly AzureServiceTokenProvider _azureServiceTokenProvider;
+        private readonly ILogger<AzureAccessTokenProvider> _logger;
+
+        public AzureAccessTokenProvider(ILogger<AzureAccessTokenProvider> logger)
+        {
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _azureServiceTokenProvider = new AzureServiceTokenProvider();
+            _logger = logger;
+        }
+
+        public string AccessTokenProviderType => "azure";
+
+        public async Task<string> GetAccessTokenForResourceAsync(Uri resourceUri, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(resourceUri, nameof(resourceUri));
+
+            string accessToken = await _azureServiceTokenProvider.GetAccessTokenAsync(resourceUri.ToString(), cancellationToken: cancellationToken);
+            if (accessToken == null)
```

Comment:
should it throw exception if it fails to get the access token instead of null?

Reply:
Changed to throw an exception when we don't get back an access token